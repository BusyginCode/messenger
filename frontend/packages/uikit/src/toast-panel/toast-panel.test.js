import React from 'react'
import { shallow } from 'enzyme'
import ToastPanel from './toast-panel'

describe('ToastPanel', () => {
  it('renders without toasts', () => {
    const toasts = []

    const panel = shallow(<ToastPanel toasts={toasts} />)

    expect(panel).toMatchSnapshot()
  })

  it('renders list of toasts', () => {
    const svg = <svg />
    const toasts = [
      {
        id: '1',
        icon: svg,
        content: 'Test message 1',
        intent: 'success'
      },
      {
        id: '2',
        icon: svg,
        content: 'Test message 2',
        intent: 'warning'
      }
    ]

    const panel = shallow(<ToastPanel toasts={toasts} />)

    expect(panel).toMatchSnapshot()
  })
})
