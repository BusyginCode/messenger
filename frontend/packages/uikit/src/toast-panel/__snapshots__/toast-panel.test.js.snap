// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastPanel renders list of toasts 1`] = `
<div
  className="toastPanel"
>
  <CSSTransitionGroup
    transitionAppear={true}
    transitionAppearTimeout={240}
    transitionEnter={true}
    transitionEnterTimeout={240}
    transitionLeave={true}
    transitionLeaveTimeout={200}
    transitionName={null}
  >
    <Toast
      className="toast"
      content="Test message 1"
      icon={<svg />}
      id="1"
      intent="success"
      key="1"
    />
    <Toast
      className="toast"
      content="Test message 2"
      icon={<svg />}
      id="2"
      intent="warning"
      key="2"
    />
  </CSSTransitionGroup>
</div>
`;

exports[`ToastPanel renders without toasts 1`] = `
<div
  className="toastPanel"
>
  <CSSTransitionGroup
    transitionAppear={true}
    transitionAppearTimeout={240}
    transitionEnter={true}
    transitionEnterTimeout={240}
    transitionLeave={true}
    transitionLeaveTimeout={200}
    transitionName={null}
  />
</div>
`;
